### PR TITLE
Implement add to cart bar

### LINKED ISSUE #1460: Implement product page add to cart bar
Implement add to cart bar according to [mockup](https://www.figma.com/file/jRG4EyQXRDMGfyLqut08KP/iFixit---Working-files?node-id=4581%3A47483&t=etNQxH5K1Jws6dKx-4)

Screenshot 2023-03-10 at 17 56 52

### PR DESCRIPTION
**Pull request by dhmacs**
closes #1460 

## Changelog

- Implement add to cart bar
- Refactor of the add to cart component

## QA

1. Open a [product page preview](https://react-commerce-git-implement-product-page-add-to-ae2b72-ifixit.vercel.app/products/repair-business-toolkit)
2. Verify that the add to cart bar is implemented according to [mockup](https://www.figma.com/file/jRG4EyQXRDMGfyLqut08KP/iFixit---Working-files?node-id=4581%3A47483&t=etNQxH5K1Jws6dKx-4)
    Screenshot 2023-03-14 at 10 29 10
3. Verify that the add to cart bar is responsive
4. Verify that clicking "add to cart" on the bar takes you back to the add to cart section

The behaviour of the bar has been validated with @mmarcon91

### CODE REVIEW

**Comment by dhmacs** [comment]
dev_block 🛑 on figuring out how to setup `IntersectionObserver` with Jest

**Comment by dhmacs** [comment]
un_dev_block 🏎️

**Comment by mmarcon91** [comment]
@davidrans @sterlinghirsh Is it possible to set up an A/B test with true/false visibility of the add to cart bar to see if it helps conversions?

**Comment by jordycosta** [comment]
QA 🟢 

The **Add to cart** bar is responsive on various view-ports ✅ 
Clicking on the **Add to cart** button on the bar moves the UI up to the main **Add to cart** button ✅ 

---

**Demo:**


https://user-images.githubusercontent.com/95656772/225082744-54480e06-6258-47f7-b123-f6b830f6f08f.mp4

https://user-images.githubusercontent.com/95656772/225082750-9e83139b-b54e-4b32-b375-08452c8010aa.mp4

**Comment by davidrans** [comment]
> Is it possible to set up an A/B test with true/false visibility of the add to cart bar to see if it helps conversions?

That's a great idea, and we should figure out how to do it. Maybe with Matomo?

File: frontend/components/ui/Rating.tsx [hunk 1]
```diff
@@ -23,14 +24,17 @@ export const Rating = (props: RatingProps) => {
             } else if (halfStarsValue >= i * 2 - 1) {
                appearance = RatingStarAppearance.Half;
             }
-            return <RatingStar key={i} appearence={appearance} />;
+            return (
+               <RatingStar key={i} size={props.size} appearence={appearance} />
```
Comment (sterlinghirsh) [inline frontend/components/ui/Rating.tsx:28]: This isn't new, but I just noticed that the prop on RatingStart `appearence` is misspelt. The variable we pass is correct (appearance).

**Review by sterlinghirsh — APPROVED** [review]
CR :zap: I'm not sure how we can set up an A/B test on this. I don't think we've done one before on react-commerce, but I'd be interested to try it. @dhmacs do you have any ideas on how we'd do it? I imagine we'd have to consider how the page will be cached. If it's quick to set up, then we can do it here. Otherwise, I might suggest merging this and then figuring out the A/B test setup later. We can always test disabling the bar after it's live.

**Comment by sterlinghirsh** [comment]
Actually since this is approved, I'm going to merge it as-is and we can figure out the A/B test later.

**Comment by sterlinghirsh** [comment]
I made issue #1468 to investigate A/B testing.

**Pull request merged**